### ISSUE TITLE
Reverse OPSIN?

### ISSUE THREAD

**Issue by simonmb** [dan2097/opsin#207]
Would it be difficult to reverse OPSIN? So to generate names from the structure like ChemAxon KNIME is able to?

I found the following: https://jcheminf.biomedcentral.com/articles/10.1186/s13321-021-00512-4
But it's a neural net, instead of a rule based system.

**Comment by dan2097** [comment]
>Would it be difficult to reverse OPSIN? 

Yes, my understanding is that the reverse would require almost entirely different code. The only synergies I can immediately think of is that the dictionary of trivial names and fused ring numbering code would be potentially useful. 
All of the relatively complete structure to name implementations I'm aware of are closed source. ChemDoodle does have an implementation, but I believe this is part of the closed source Java backend rather than the open soure ChemDoodle Web Components.

See also #122

**Comment by HinnerkHD** [comment]
autonom (<https://pubs.acs.org/doi/10.1021/ci00002a007>) is / was free. but it is no longer supported.

**Comment by iclkevin** [comment]
@dan2097 is correct, the ChemDoodle IUPAC naming algorithms are proprietary and are exposed through a Java API. The ChemDoodle Web Components has AJAX hooks into the Java API.

@simonmb, if you need any help with IUPAC naming, and are ok with a commercial solution, iChemLabs is always happy to help. Please just let me know.

**Comment by flange-ipb** [comment]
I'd like to make you aware of https://github.com/lamalab-org/openclatura as a viable solution for IUPAC name generation.

**Comment by simonmb** [comment]
Nice, thanks!
